src/control.cc

```diff
@@ -1443,7 +1443,7 @@ bool control_conn_t::data_ready() noexcept
     }
     
     // complete packet?
-    while (rbuf.get_length() >= chklen) {
+    while (rbuf.get_length() >= chklen && !bad_conn_close) {
         try {
             if (!process_packet()) {
```

Comment:
The check for bad_conn_close should be here (not in the `while` condition).

    if (!process_packet() || bad_conn_close) {